src/SubPool.sol

```diff
@@ -0,0 +1,157 @@
+    error SubPool__ExitDelayNotElapsedYet();
+    error SubPool__CollateralTokenAlreadyInitialized();
+    error SubPool__InsufficientETH();
+    error SubPool__CannotBillAfterExitDelay();
+    error SubPool__InvalidWithdraw();
+
+    /// @dev cannot be immutable because that allows for multiple pools per solver address.
+    address public collateralToken;
+    address public immutable COW;
+    ISubPoolFactory public immutable factory;
+
+    uint256 collateralDue;
+    uint256 cowDue;
+    uint256 ethDue;
+
+    constructor(address _owner, address _cow) {
+        factory = ISubPoolFactory(msg.sender);
+        COW = _cow;
+
+        // rely the owner
+        _addOwner(_owner);
+    }
+
+    function initializeCollateralToken(address token) external {
+        if (collateralToken != address(0)) revert SubPool__CollateralTokenAlreadyInitialized();
+        collateralToken = token;
+    }
+
+    modifier onlyFactory() {
+        if (msg.sender != address(factory)) {
+            revert SubPool__OnlyFactory();
+        }
+        _;
+    }
+
+    /// @notice Signal the intent to announce exit the solver pool.
+    function announceExit() external auth {
+        factory.announceExit();
+    }
+
+    /// @notice Execute the exit after the exit delay has elapsed.
+    function exit() external auth {
+        uint256 exitTimestamp = factory.exitTimestamp(address(this));
+        // can skip the 0 check i.e. pool not announced an exit check because block.timestamp
+        // will never be < 0
+        if (block.timestamp < exitTimestamp) revert SubPool__ExitDelayNotElapsedYet();
+
+        uint256 collateralBalance = ERC20(collateralToken).balanceOf(address(this));
+        uint256 cowBalance = ERC20(COW).balanceOf(address(this));
+        SafeTransferLib.safeTransfer(collateralToken, msg.sender, collateralBalance);
+        SafeTransferLib.safeTransfer(COW, msg.sender, cowBalance);
+        SafeTransferLib.safeTransferETH(msg.sender, address(this).balance);
+
+        factory.exitPool();
+    }
+
+    /// @notice withdraw arbitrary tokens
+    /// @dev Can only withdraw non cow and non collateral tokens while the pool is active or in exit delay.
+    ///      When the pool's exit delay has elapsed it can withdraw any token and ether balance.
+    function withdrawTokens(address[] calldata tokens) external auth {
+        uint256 exitTimestamp = factory.exitTimestamp(address(this));
+        bool exitElapsed = exitTimestamp != 0 && block.timestamp >= exitTimestamp;
+
+        if (exitElapsed) {
+            for (uint256 i = 0; i < tokens.length;) {
+                SafeTransferLib.safeTransfer(tokens[i], msg.sender, ERC20(tokens[i]).balanceOf(address(this)));
+                unchecked {
+                    ++i;
+                }
+            }
+            uint256 ethBalance = address(this).balance;
+            if (ethBalance > 0) {
+                SafeTransferLib.safeTransferETH(msg.sender, address(this).balance);
```

Comment:
Can anyone transfer out ETH without exiting fully? Would be useful to harvest the MEVBlocker rewards and convert them into collater token.

Reply:
Not in the current implementation, they can only transfer tokens that are not collateral, COW, or native ETH

Reply:
We can implement conditional orders when eth balance > X, then swap for collateral token, but feels like unnecessary complexity. Should eth balance just be withdrawable even when the pool hasn't exited like all non-collateral tokens?

Reply:
I think we should be taking an opinionated stance here. We already grant significant latitude to solvers joining with different types of collateral (increasing load on off-chain infrastructure). Maintaining ETH is fine IMO (and increases diversification).

Reply:
I guess the factory owner can bill any ETH, convert it into collateral and deposit it on behalf of the subpool owner. I believe this is sufficient.

Reply:
no change here